test/src/test/kotlin/graphql/nadel/tests/next/GraphQLServiceExecution.kt

```diff
@@ -41,7 +45,16 @@ class GraphQLServiceExecution(
                         errors = spec["errors"] as MutableList<MutableMap<String, Any?>>? ?: mutableListOf(),
                         extensions = spec["extensions"] as MutableMap<String, Any?>? ?: mutableMapOf(),
                         incremental = it.incremental,
-                        incrementalItemPublisher = it.incrementalItemPublisher,
+                        incrementalItemPublisher = it.incrementalItemPublisher
+                            .asFlow()
+                            .catch { e ->
+                                onServiceResponse(input, success = null, e)
+                                throw e
+                            }
+                            .onEach { result ->
+                                onServiceResponse(input, result?.toSpecification(), failure = null)
```

Comment:
Should we have a different calback for incremental responses? Or at least add a flag in the arguments - `incremental: boolean`